frontend/src/components/modals/AddingTraineeModal.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import ModalLayout from "./ModalLayout";
+import InputField from "../ui/InputField";
+import Button from "../ui/Button";
+import { useForm } from "react-hook-form";
+import {
+  useCreateTraineeMutation,
+  useGetAllCoachesQuery,
+} from "../../features/user/apiSlice";
+import Loader from "../ui/Loader";
+import Alert from "../ui/Alert";
+
+const AddingTraineeModal = ({
+  closePopup,
+  jwt,
+}: {
+  closePopup: () => void;
+  jwt: string;
+}) => {
+  const [createTrainee, { isError, isLoading, error, isSuccess }] =
+    useCreateTraineeMutation();
+  const coachesData = useGetAllCoachesQuery({ jwt, query: "" });
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const onSubmit = async (data: any) => {
+    if (!data?.coach) {
+      delete data.coach;
+    }
+    const result = await createTrainee({
+      jwt,
+      body: { ...data, role: "TRAINEE" },
+    });
+  };
+  let errorMessage: any = errors?.name?.message || errors?.email?.message;
+  if (error?.data?.code === 11000) {
```

Comment:
What does 11000 code means ?

Reply:
This is to catch duplicate errors for either email or name. 11000 is code specifically for duplicate error